libnavigator/src/androidTest/java/com/mapbox/navigation/navigator/internal/NavigatorTest.kt

```diff
@@ -0,0 +1,123 @@
+package com.mapbox.navigation.navigator.internal
+
+import androidx.annotation.RawRes
+import androidx.test.espresso.Espresso
+import androidx.test.platform.app.InstrumentationRegistry
+import com.mapbox.api.directions.v5.DirectionsCriteria
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.navigator.internal.util.readRawFileText
+import com.mapbox.navigation.navigator.internal.util.runOnMainSync
+import com.mapbox.navigator.CacheFactory
+import com.mapbox.navigator.CacheHandle
+import com.mapbox.navigator.ConfigFactory
+import com.mapbox.navigator.ConfigHandle
+import com.mapbox.navigator.Navigator
+import com.mapbox.navigator.NavigatorConfig
+import com.mapbox.navigator.ProfileApplication
+import com.mapbox.navigator.ProfilePlatform
+import com.mapbox.navigator.Routes
+import com.mapbox.navigator.SettingsProfile
+import com.mapbox.navigator.TilesConfig
+import org.junit.Assert.assertTrue
+import org.junit.Test
+
+class NavigatorTest {
+
+    private companion object {
+        @RawRes
+        private val DIRECTIONS_RESPONSE =
+            com.mapbox.navigation.navigator.test.R.raw.directions_response_two_leg_route
+    }
+
+    @Test(timeout = 10000)
+    fun successfulSetRoutesToNN() {
+        Espresso.onIdle()
+        runOnMainSync {
+            val navigator = provideNavigator()
+            val (routes, routeOptions) = provideDirectionsRouteAndRouteOptions()
+            navigator.setRoutes(
+                Routes(
+                    routes.mapToDirectionsResponse(routeOptions).toJson(),
+                    0,
+                    0,
+                    routeOptions.toUrl("pk.**").toString()
+                )
+            ) { expected ->
+                assertTrue(expected.isValue)
+            }
```

Comment:
How does the test wait for the callback? Can if finish before callback is called?

Reply:
https://github.com/mapbox/mapbox-navigation-android/pull/5500#discussion_r818828008

Reply:
I put a breakpoint to the line 49 and it's never triggered. Does it work for you?